streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -217,15 +216,17 @@ public void makeReady(final Map<String, InternalTopicConfig> topics) {
     /**
      * Check the existing topics to have correct number of partitions; and return the remaining topics that needs to be created
      */
-    private Set<String> validateTopics(final Set<String> topicsToValidate,
-                                       final Map<String, InternalTopicConfig> topicsMap) {
+    private Set<String> validateTopics(final Set<String> topicsToValidate, final Map<String, InternalTopicConfig> topicsMap) {
+        if (!topicsMap.keySet().containsAll(topicsToValidate)) {
+            throw new IllegalStateException("The topics map " + topicsMap + " does not contain all the topics " +
```

Comment:
req: Can we log an error with the two topics sets (`topicsMap.keySet` and `topicsToValidate`), and then just say what happened in the exception message? Logging the entire map in the exception message will be pretty ugly